src/blueprint/azext_blueprint/custom.py

```diff
@@ -0,0 +1,580 @@
+            body['description'] = blueprint_properties['description']  # str
+        if 'targetScope' in blueprint_properties:
+            body['target_scope'] = blueprint_properties['targetScope']  # str
+        else:
+            body['target_scope'] = 'subscription'
+        if 'parameters' in blueprint_properties:
+            body['parameters'] = blueprint_properties[
+                'parameters']  # dictionary
+        if 'resourceGroups' in blueprint_properties:
+            body['resource_groups'] = blueprint_properties[
+                'resourceGroups']  # dictionary
+        blueprint_response = client.create_or_update(
+            scope=scope, blueprint_name=blueprint_name, blueprint=body)
+
+    # delete old artifacts
+    artifacts = artifact_client.list(scope=scope, blueprint_name=blueprint_name)
+    for artifact in artifacts:
+        artifact_client.delete(scope=scope,
+                               blueprint_name=blueprint_name,
+                               artifact_name=artifact.name)
+
+    for filename in os.listdir(os.path.join(os.path.expanduser(input_path), 'artifacts')):
+        artifact_name = filename.split('.')[0]
+        filepath = os.path.join(os.path.expanduser(input_path), 'artifacts', filename)
+        with open(filepath) as artifact_file:
+            artifact = json.load(artifact_file)
+            artifact_client.create_or_update(scope=scope,
+                                             blueprint_name=blueprint_name,
+                                             artifact_name=artifact_name,
+                                             artifact=artifact)
+
+    return blueprint_response
+
+
+def create_blueprint(cmd,
+                     client,
+                     blueprint_name,
+                     management_group=None,
+                     scope=None,
+                     target_scope='subscription',
+                     display_name=None,
+                     description=None,
+                     parameters=None):
+    body = {}
+    body['display_name'] = display_name  # str
+    body['description'] = description  # str
+    body['target_scope'] = target_scope  # str
+    body['parameters'] = json.loads(
+        parameters) if parameters is not None else {}  # dictionary
+
+    return client.create_or_update(scope=scope,
+                                   blueprint_name=blueprint_name,
+                                   blueprint=body)
+
+
+def update_blueprint(cmd,
+                     client,
+                     blueprint_name,
+                     management_group=None,
+                     scope=None,
+                     description=None,
+                     parameters=None):
+    body = client.get(scope=scope, blueprint_name=blueprint_name).as_dict()
+    if description is not None:
+        body['description'] = description  # str
+    if parameters is not None:
+        body['parameters'] = json.loads(parameters)  # dictionary

```

Comment:
json load exception catch/rethrow?

Reply:
Moved the json.loads() logic to validator, parameters is of CLIArgumentType, this type already handles json parse error. It will not reach the code here in case of json format error.